util/encoding/encoding.go

```diff
@@ -444,3 +453,255 @@ func DecodeBytes(b []byte) ([]byte, []byte) {
 		b = b[i+2:]
 	}
 }
+
+// DecodeBytesDecreasing decodes a []byte value from the input buffer
+// which was encoded using EncodeBytesDecreasing. The remainder of the
+// input buffer and the decoded []byte are returned.
+func DecodeBytesDecreasing(b []byte, r []byte) ([]byte, []byte) {
```

Comment:
Could you eliminate this duplication by making a decodeBytesHelper that takes the escape characters as arguments?